core/setup.py

```diff
@@ -51,6 +51,7 @@
         "ray[default]>=2.10.0,<2.32",  # sync with common/src/autogluon/common/utils/try_import.py
     ],
     "raytune": [
+        "pyarrow>=17.0.0",  # cap Pyarrow to fix source installation - https://github.com/autogluon/autogluon/issues/4519
```

Comment:
I think 17.0.0 is the latest version. Is it safer to cap it to a lower version? I tested that capping to `12.0.0` also works.

Reply:
I tested all python versions on 17.0 and they passed.
@Innixma suggested that we keep this cap

Reply:
Also cap it with upperbound as well in case future release broke autogluon

Reply:
@tonyhoo since raytune depends on pyarrow, this is an implicit dependency, and we rely on ray to set a proper upperbound. This is only installed when paired with ray to limit the version ranges pip will search through. Therefore, I don't think we need an upperbound.